spec/special-cases.js

```diff
@@ -12,6 +13,31 @@ describe("special cases", () => {
       expect(html).to.equal("<div></div>");
     });
   });
+  it("renders empty comments for components that return null", () => {
+    const NullComponent = () => null;
+
+    return render(<NullComponent />)
+      .toPromise()
+      .then(html => expect(html).to.equal("<!-- react-empty: 1 -->"));
+  });
+  it("does not render text comments for children without siblings", () => {
+    const TextComponent = () => "some text";
+
+    return render(<TextComponent />).toPromise()
+      .then(html => expect(html).to.equal("some text"));
+  });
+  it("renders text comments for children with siblings", () => {
+    const TextWithSiblingsComponent = () => <div id="root"><div>child 1</div>child 2</div>;
+
+    const expected =
+      "<div data-reactid=\"2\">child 1</div><!-- react-text: 3 -->child 2<!-- /react-text -->";
+
+    return render(<TextWithSiblingsComponent />)
+      .toPromise()
+      .then(cheerio.load)
+      .then($ => $("#root").html())
+      .then(html => expect(html).to.equal(expected));
```

Comment:
I could include the wrapper. The reason I didn't was because it has the checksum and seemed like a weird thing to embed into an unassociated unit test but it's not a big deal. I'll remove the dependency.